frontend/src/logger/RemoteLogger.ts

```diff
@@ -96,9 +102,27 @@ export class Logger implements ILogger {
 
   private buildMessage(
     level: LogLevel,
-    message: string,
+    payload: Error | string,
     extra?: Record<string, unknown>,
   ): LogEntry {
-    return {message, level, extra}
+    if (!payload) {
+      return {message: DEAFULT_MESSAGE, level, extra}
+    }
+
+    if (payload instanceof Error) {
+      extra = {
+        ...extra,
+        trace: payload.stack,
```

Comment:
suggestion(non-blocking): I'd call this field `stacktrace` since trace is vague and makes me think of tracing